src/ChainSafe.Gaming.Chainlink.Lootboxes/LootboxService.cs

```diff
@@ -147,7 +153,29 @@ public async Task<bool> IsOpeningLootbox()
 
         public async Task<uint> OpeningLootboxType()
         {
-            throw new NotImplementedException();
+            var playerAddress = await this.GetCurrentPlayerAddress();
+
+            // This response is actually very different from all the others since it returns several components
+            var response = (List<ParameterOutput>)(await this.contract.Call("getOpenerRequestDetails", new object[] { playerAddress }))[0];
+
+            var address = (string)response[0].Result;
+            var unitsToGet = (BigInteger)response[1].Result;
+
+            if (!string.IsNullOrEmpty(address) && unitsToGet == 0)
+            {
+                // we can early return here, but, it's not necessary since unitstoget will be 0 regardless and this
+                // call will fulfill every request that's been missing.
+                await this.contract.Send(
+                    "recoverBoxes",
```

Comment:
Not sure what's happening here

Reply:
Based on Oleksii's feedback: 
`If the result is not empty (opener is not zero address) and unitsToGet == 0, then user needs to recover failed boxes by sending a tx: recoverBoxes(address opener)` 
I am sending this as a transaction. I haven't encountered any situation in which the condition above would be fulfilled tbh.

Reply:
I see. If it's fixing broken lootboxes then we should put it in a separate method, I think. Because this one is kinda "pure", it just returns you the type id of the lootbox that's currently being opened. What do you think?

Reply:
I kinda agree, but the issue I'm having with that Oleksii said that we need to check the response in there. I mean we can maybe pass the unitsToGet back to the calling method and some stuff there if it's equal to zero.